src/engine/sequencer.js

```diff
@@ -54,28 +49,18 @@ Sequencer.prototype.stepThreads = function (threads) {
             var activeThread = threads[i];
             if (activeThread.status === Thread.STATUS_RUNNING) {
                 // Normal-mode thread: step.
-                this.stepThread(activeThread);
+                this.startThread(activeThread);
             } else if (activeThread.status === Thread.STATUS_YIELD) {
                 // Yield-mode thread: check if the time has passed.
-                YieldTimers.resolve(activeThread.yieldTimerId);
-                numYieldingThreads++;
+                if (!YieldTimers.resolve(activeThread.yieldTimerId)) {
+                    numYieldingThreads++;
+                }
```

Comment:
At first glance it looks like a single yielding thread could cause `numYieldingThreads` to grow until it's big enough to exceed `threads.length`, as long as there are other threads which want to keep executing and the work time hasn't run out. Is there something here which prevents that?